### PR TITLE
Adding C++ tools for compilation, linking and debug

### PR DESCRIPTION
**Pull request by matheecs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,14 @@ Use **cache** to speed up the compilation. If you compile the same project again
 cmake .. -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
 ```
 
+Use [Clang](https://clang.llvm.org/) & [Ninja](https://github.com/ninja-build/ninja) to speed up the compilation.
```
Comment (stephane-caron) [inline README.md:16]: Ninja is a build system altogether. As far as I can tell Simple-Robotics software mostly uses CMake. Do we want to also include in the list software that is not used in Simple-Robotics?

1. If yes, we can start a separate "Build system" section. We can also add Bazel in there :wink: @EtienneAr @fabinsch @jcarpent 
2. If no, we need to clarify the scope of this repository in its readme.

I'd be in favor of 2, as if we go for 1. the list may progressively include more and more pointers to things that are not actually in use (at least in Simple-Robotics).

File: README.md [hunk 2]
```diff
@@ -37,6 +45,31 @@ run
...
+
+```bash
+ulimit -c unlimited
+```
+Then, view the backtrace in gdb
+
+```bash
+gdb path/to/my/executable path/to/coredumpfile
+```
+See [here](https://askubuntu.com/questions/1349047/where-do-i-find-core-dump-files-and-how-do-i-view-and-analyze-the-backtrace-st) for further details.
```
Comment (stephane-caron) [inline README.md:60]: :+1:

File: README.md [hunk 3]
```diff
@@ -37,6 +45,31 @@ run
...
+### Usage Backward 
+
+[Backward](https://github.com/bombela/backward-cpp) is a beautiful stack trace pretty printer for C++. You need to compile your project with generation of debug symbols enabled, usually `-g` with clang++ and g++. Add the following code to the source file
+
+```c++
+#include <backward.hpp>
+namespace backward {
+  backward::SignalHandling sh;
+}
+```
```
Comment (stephane-caron) [inline README.md:71]: :+1:

File: README.md [hunk 4]
```diff
@@ -141,6 +174,15 @@ valgrind --leak-check=yes myprog arg1 arg2
 
 Check [here](https://stackoverflow.com/questions/5134891/how-do-i-use-valgrind-to-find-memory-leaks) and [doc](https://valgrind.org/docs/manual/quick-start.html) for further explanation.
 
+### AddressSanitizer
+
+[AddressSanitizer](https://github.com/google/sanitizers) (aka ASan) is a memory error detector for C/C++. For cmake, you can add link option in `CMakeLists.txt`
```
Comment (stephane-caron) [inline README.md:179]:
```suggestion
[AddressSanitizer](https://github.com/google/sanitizers/wiki/AddressSanitizer) (aka ASan) is a memory error detector for C/C++. For cmake, you can add link option in `CMakeLists.txt`
```

**Review by stephane-caron — COMMENTED** [review]
Thank you for these suggestions, overall I think they make great additions :smiley:

Regarding the build system/compiler/linker I'd just have one question to other maintainers about the scope of this list. Other than this looks :+1: to me.

**Comment by jcarpent** [comment]
@stephane-caron @matheecs What is the status of this PR?

File: README.md (hunk 1 above)
Comment (matheecs) [reply README.md:16]: Ninja works similarly to Make; Bazel works similarly to Make. Ninja is more like Make than CMake.

File: README.md (hunk 4 above)
Comment (matheecs) [reply README.md:179]: Thanks for the suggestion.

File: README.md (hunk 1 above)
Comment (stephane-caron) [reply README.md:16]: Given the scope of the repo:

> The idea is to gather some tools that we use to develop high performance code. We should provide information about how to install and use these tools.

I'd either remove this line (Ninja is not used in Simple-Robotics right now) or provide information about how to install and use it, for example as a CMake backend.

**Comment by stephane-caron** [comment]
There is one pending question on either providing usage info on Ninja or not including it in the PR, other than this LGTM :+1:

**Pull request merged**